src/lsm/lsm_manager.c

```diff
@@ -658,6 +658,14 @@ __wt_lsm_manager_push_entry(WT_SESSION_IMPL *session,
 	manager = &S2C(session)->lsm_manager;
 
 	/*
+	 * Don't allow any work units unless a tree is active, this avoids
+	 * races on shutdown between clearing out queues and pushing new
+	 * work units.
+	 */
+	if (!F_ISSET(lsm_tree, WT_LSM_TREE_ACTIVE))
```

Comment:
This is the bug fix. The problem is that processing of work units doesn't hold a tree reference (it bumps the work queue reference count instead). Ordering in just the right way could leave the reference counts at 0 which lets drop proceed, but a work unit is in the process of being pushed.